src/meta/src/stream/source_manager.rs

```diff
@@ -743,21 +750,27 @@ impl SourceManager {
             ConnectorSourceWorkerHandle {
                 handle,
                 sync_call_tx,
-                splits: current_splits_ref,
+                splits,
                 enable_scale_in,
             },
         );
         Ok(())
     }
 
+    /// Used when registering new sources.
     async fn create_source_worker(
         connector_client: Option<ConnectorClient>,
         source: &Source,
         managed_sources: &mut HashMap<SourceId, ConnectorSourceWorkerHandle>,
         force_tick: bool,
         metrics: Arc<MetaMetrics>,
     ) -> MetaResult<()> {
-        let current_splits_ref = Arc::new(Mutex::new(SharedSplitMap { splits: None }));
+        tracing::info!("spawning new watcher for source {}", source.id);
+
+        let splits = Arc::new(Mutex::new(SharedSplitMap { splits: None }));
+        let current_splits_ref = splits.clone();
+        let source_id = source.id;
+
```

Comment:
The changes are to make `create_source_worker` and `create_source_worker_async` more alike. I also failed to find a nicer way to unify the two...